### PR TITLE
WIP: Tests

### PR DESCRIPTION
**Pull request by koca**
## Description
Tests #9 

## Motivation and Context
We'll be confident :D

## Component list

Feel free to add a component to the list.

- [x] Avatar
- [x] AvatarGroup
- [x] Box
- [x] PseudoBox
- [x] Button
- [x] ButtonGroup
- [x] CloseButton
- [x] Divider
- [x] Icon
- [x] IconButton
- [x] Text
- [x] Alert
- [x] Badge
- [ ] Toast
- [x] Flex
- [x] Heading
- [ ] ControlBox
- [ ] Stack
- [ ] Modal
- [ ] Drawer
- [ ] AlertDialog
- [x] PseudoBox
- [x] BreadCrumb
- [ ] Tooltip
- [x] CircularProgress
- [x] Code
- [x] Grid
- [x] Progress
- [ ] Menu
- [ ] Popover
- [x] Collapse
- [x] Accordion
- [ ] AspectRatio
- [ ] Image
- [ ] Input
- [x] FormControl
- [ ] Radio
- [ ] RadioGroup
- [ ] Tag
- [x] Switch
- [ ] Slider
- [x] Checkbox
- [ ] CheckboxGroup
- [x] Fragment

### CODE REVIEW

File: packages/kiwi-core/rollup.config.js [hunk 1]
```diff
@@ -79,6 +79,7 @@ const capitalize = (s) => {
 
 const components = fs.readdirSync(baseFolder)
   .filter((f) => fs.statSync(path.join(baseFolder, f)).isDirectory())
+  .filter(f => f !== 'test-utils')
```
Comment (koca) [inline packages/kiwi-core/rollup.config.js]: We could make package for tests-utils? What do you think @codebender828

**Comment by koca** [comment]
For the simple components like Flex or Box etc. we could test the css like using jest-emotion serializer. But it'll add so much css style outputs to all the snapshots...
Hmm how should we go about it? 🤔 cc @codebender828

File: packages/kiwi-core/src/test-utils/test-utils.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import '@testing-library/jest-dom/extend-expect'
```
Comment (codebender828) [inline packages/kiwi-core/src/test-utils/test-utils.js]: I think these can be moved to their own `tests` directory in the package root. that way the `src` directory only keeps the components.

File: stories/10-Toast.stories.js [hunk 3]
```diff
@@ -1,9 +1,26 @@
 import { storiesOf } from '@storybook/vue'
 import { Button } from '../packages/kiwi-core/src'
+import { colorModeObserver } from '../packages/kiwi-core/src/utils/color-mode-observer'
+
+const watch = {
```
Comment (codebender828) [inline stories/10-Toast.stories.js]: 😄

File: packages/kiwi-core/src/test-utils/test-utils.js (hunk 2 above)
Comment (koca) [reply packages/kiwi-core/src/test-utils/test-utils.js]: moved ^^

File: packages/kiwi-core/rollup.config.js (hunk 1 above)
Comment (koca) [reply packages/kiwi-core/rollup.config.js]: moved test-utils to tests folder

File: packages/kiwi-core/src/Breadcrumb/tests/Breadcrumb.test.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          <BreadcrumbLink href="#">Breadcrumb 2</BreadcrumbLink>
+          <BreadcrumbSeparator color="firebrick" font-size="20px" font-weight="bold" />
+        </BreadcrumbItem>
+      </Breadcrumb>
+    </div>
+   `,
+    ...props
+  }
+  return render(base, {}, vue => {
+    vue.use(CompositionApi)
```
[the remark this answers is not in the document]
Comment (koca) [reply packages/kiwi-core/src/Breadcrumb/tests/Breadcrumb.test.js]: Ok. Updated the tests ^^

File: packages/kiwi-core/src/Button/tests/button.test.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+it('should display spinner and hide text', () => {
+  const { getByTestId, container } = renderComponent({ template: `<Button isLoading data-testid="btn">Button</Button>` })
+  const button = getByTestId('btn')
+
+  expect(button).toHaveAttribute('disabled')
+  expect(button).toHaveAttribute('aria-disabled', 'true')
+
+  // TODO: find a way to easily grab the spinner element
+  // Maybe? expect(getByTestId('Spinner')).toBeInTheDocument()
+  expect(container.querySelector('button > div')).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (koca) [reply packages/kiwi-core/src/Button/tests/button.test.js]: Sounds good ^^

File: packages/kiwi-core/src/Box/tests/Box.test.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import Box from '../'
+import { render, defaultProviders } from '@/tests/test-utils'
```
Comment (codebender828) [inline packages/kiwi-core/src/Box/tests/Box.test.js]: Nice work here! ⚡✌🏽

**Pull request merged**